pkg/events/derive/hidden_kernel_module.go

```diff
@@ -81,13 +129,16 @@ func clearMap(bpfMap *bpf.BPFMap) error {
 	return nil
 }
 
+func GetWakeupChannelRead() <-chan ScanRequest {
```

Comment:
See, now anything can read from this module's messaging. So another module calls this and starts extracting messaging and now there's a conflict between it and `tracee.go`.
There's likely a way to avoid cross module messaging here.

Reply:
I don't see any reason why it would be called elsewhere (I'm guessing the design is what you're worried about), but I could move everything to hidden_kernel_module.go and make the channel "private". Assuming I can move the uprobed function (triggerKernelModuleSeeker) from tracee.go to my derived event, my only problem is the kernelSymbols which I need to read periodically, that is under Tracee object.

Reply:
IMO: lets have the logic working like it is, with no known issues after review, and then we should move this channel to something "generic" for all the event derivation logic in another PR (something like a "workqueue" logic for a derived events "processEvents" step (I mentioned this in a previous comment, based on a feedback already given by Nadav)).

The code is already too complex to include something like this now, IMHO: we should concentrate in having this work the best way, spin off code from big source files to their own files (so we can see all triggering related functions in a single place) and document it better.

I would move all the uprobe hook functions and methods from tracee object:

![image](https://user-images.githubusercontent.com/7395852/233762486-f536925a-c4ee-48c0-881f-beaa2dd2b083.png)

into its own file under ebpf package directory, for example.